Add bindings to cast rays or project points on a single collider

sebcrozet opened the pull request:
This adds methods to the `Collider` object: `containsPoint, projectPoint, castRay, castRayAndGetNormal, intersectsRay`.
This also fixes the ordering of joint variants (fix #68).

jcyuan commented:
fix need to be applied for **Collider.ts** file:

1, import error
```typescript
import { Ray, RayIntersection } from "./ray";
import { PointProjection } from "./point";
```

2, `castRay` method is missing
```typescript
 * @returns toi between this collider and the ray, or `-1` if there is no intersection.
     */
    public castRay(
        ray: Ray,
        maxToi: number,
        solid: boolean,
    ): number {
        let rawOrig = VectorOps.intoRaw(ray.origin);
        let rawDir = VectorOps.intoRaw(ray.dir);
        let result = this.rawSet.coCastRay(
            this.handle,
            rawOrig,
            rawDir,
            maxToi,
            solid,
        );

        rawOrig.free();
        rawDir.free();

        return result;
    }
```

jcyuan commented:
tested:

castRay
castRayAndGetNormal
intersectsRay
projectPoint
containsPoint

all pass.

btw: JointType enum not match problem has been fixed, but velocity for motor still has bug.

jcyuan commented:
and a trivil thing:

in **collider.rs**
```rust
use crate::geometry::{
    RawColliderSet, RawPointProjection, RawRayIntersection, RawShape, RawShapeType,
};
```
useless comma found at the tail.

sebcrozet commented:
Thank you for the review. I addressed your comments.
I didn’t remove the useless comma because that’s part of Rust’s coding style (and it’s automatically added by the `rustfmt`).
